### ISSUE TITLE
0.4.3.88 PDH counter with rrd buffer will return oldest value from buffer as current/actual value

### ISSUE THREAD

**Issue by MilanDadok** [mickem/nscp#87]
[/settings/system/windows/counters/LogicalDiskCDiskTime]
; COUNTER - The counter to check
counter = \LogicalDisk(C:)\% Disk Time

; COLLECTION STRATEGY - The way to handled values when collecting them: static means we keep the last known value, rrd means we store values in a buffer from which you can retrieve the average
collection strategy = round robin

; BUFFER SIZE - Size of buffer (in seconds) larger buffer use more memory
buffer size = 60s

after staring nscp test, there check_pdh counter=LogicalDiskCDiskTime will return 0 for first 60 second, values are returned with 60 second delay

with next patch that works's as I expected
```
diff --git a/include/pdh/pdh_collectors.hpp b/include/pdh/pdh_collectors.hpp
index d78a471..56efdf9 100644
--- a/include/pdh/pdh_collectors.hpp
+++ b/include/pdh/pdh_collectors.hpp
@@ -210,13 +210,13 @@ namespace PDH {
 				boost::shared_lock<boost::shared_mutex> lock(mutex_);
 				if (!lock.owns_lock())
 					throw PDH::pdh_exception(get_name(), "Could not get mutex");
-				return values.front();
+				return values.back();
 			}
 			virtual long long get_int_value() {
 				boost::shared_lock<boost::shared_mutex> lock(mutex_);
 				if (!lock.owns_lock())
 					throw PDH::pdh_exception(get_name(), "Could not get mutex");
-				return values.front();
+				return values.back();
 			}
 
 			virtual void update(T value) {
```

**Comment by mickem** [comment]
Thanks, how about a pull request?

**Comment by mickem** [comment]
Should be fixed in the latest nightly build (0.4.3.90)
